src/foam/strategy/BasicStrategizer.js

```diff
@@ -37,7 +37,10 @@ foam.CLASS({
           throw new RuntimeException("A desired model id must be specified.");
         }
 
-        Predicate predicate = SafetyUtil.isEmpty(target)
+        if ( strategyPredicate != null  ) {
+          predicate = (Predicate) strategyPredicate;
+        } else {
+          Predicate predicate = SafetyUtil.isEmpty(target)
```

Comment:
Does this compile? Since you use the variable on line 41 that you don't declare until line 43?